### PR TITLE
Fix appname on linux

### PR DESCRIPTION
**Pull request by ckerr**
__Rationale__

Notifications can't currently be tested in CI and the 2.0.0-beta.1 window is about to close, so this PR is a subset of the `linux-appname` branch with the API changes in `brightray/` and `atom/` (the breaking changes) but without the notification code changes, which are nonbreaking fixes that can be landed later when they are testable.

__Changes__

* Add new function `brightray::OverrideApplicationVersion()` so that version info will be in brightray's scope, and reimplement `app.setVersion()` to use it
* On Linux, implement `brightray::GetApplicationVersion()` to use the value of `app.setVersion()` if set
* On Linux, implement `brightray::GetApplicationName()` to use (1) the value of `app.setName()` if available or (2) the 'Name' entry from the app's .desktop file.
* Fix pre-existing API spelling error: s/Overriden/Overridden/ in `brightray::GetOverridenApplicationName()`

### CODE REVIEW

File: atom/browser/browser.cc [hunk 1]
```diff
@@ -97,27 +97,21 @@ void Browser::Shutdown() {
...
 
 std::string Browser::GetVersion() const {
-  if (version_override_.empty()) {
-    std::string version = GetExecutableFileVersion();
-    if (!version.empty())
-      return version;
-  }
-
-  return version_override_;
+  std::string ret = brightray::GetOverriddenApplicationVersion();
```
Comment (codebytere) [inline atom/browser/browser.cc]: (nit) think this might be `const`

File: atom/common/linux/application_info.cc [hunk 2]
```diff
@@ -2,18 +2,74 @@
...
 #include "atom/common/atom_version.h"
+#include "base/environment.h"
+#include "base/logging.h"
+#include "chrome/browser/ui/libgtkui/gtk_util.h"
+
+namespace {
+
+GDesktopAppInfo* get_desktop_app_info() {
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  const std::string desktop_id = libgtkui::GetDesktopName(env.get());
```
Comment (deepak1556) [inline atom/common/linux/application_info.cc]: Doesn't it default to `chromium-browser.desktop` https://cs.chromium.org/chromium/src/chrome/browser/ui/libgtkui/gtk_util.cc?type=cs&q=GetDesktopname&sq=package:chromium&l=132 when the `CHROME_DESKTOP` env isn't set, should we avoid this default value ?

Comment (ckerr) [reply atom/common/linux/application_info.cc]: I agree with both these points & will address them in the `linux-appname` branch

Comment (ckerr) [reply atom/common/linux/application_info.cc]: Addressed in https://github.com/electron/electron/pull/11957/commits/38dd5cee68fd5587f10de1e40b2d9291b8d8b3ba and https://github.com/electron/electron/pull/11957/commits/c8d2487747a0ffcd579c8f696a25e61beab29f00

**Pull request merged**